### PR TITLE
Creates `IBMQNoiseModel`

### LINKED ISSUE #1214: Remove `noise_model.noisy_circuit` and `NoiseModel.composite`

### PR DESCRIPTION
**Pull request by renatomello**
Closes #1214 .  

New class `IBMQNoiseModel` inherits `NoiseModel`, but allows for the creating of a noise model involving `ThermalRelaxationError`, `DepolarizingError`, and `ReadoutError`.

Noise model initialisation is done using a dictionary of parameters.

This class is meant to replace `CompositeNoiseModel` defined in `noise_model.py`.

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [x] Tests are passing.
- [x] Coverage does not decrease.
- [x] Documentation is updated.

### CODE REVIEW

**Comment by renatomello** [comment]
@rahula06 @MatteoRobbiati @AlejandroSopena this is ready for review

**Comment by rahula06** [comment]
Thanks a lot for your work on this, @renatomello :)

I've been using this branch in our application, replacing the `CompositeNoiseModel` with the `IBMQNoiseModel` and it seems to work well. I'll do the same with the latest commits as well, just in case

**Comment by renatomello** [comment]
> Thanks a lot for your work on this, @renatomello :)
> 
> I've been using this branch in our application, replacing the `CompositeNoiseModel` with the `IBMQNoiseModel` and it seems to work well. I'll do the same with the latest commits as well, just in case

Can I ask you to review this PR between today and tomorrow? I'd like your input since you're using the class in an application. And if we can merge this by tomorrow it's going to be in the next `qibo` release.

**Review by rahula06 — APPROVED** [review]
Everything looks good to me, there are no problems in application.

File: doc/source/api-reference/qibo.rst
Comment (rahula06) [inline doc/source/api-reference/qibo.rst:1135]: I think there might be a typo here in "reported noise model of for its quantum computer".

**Comment by scarrazza** [comment]
@renatomello can we conclude this later today for the next release?

**Comment by renatomello** [comment]
Only missing a second review, on my end it's done

**Review by scarrazza — APPROVED** [review]
@renatomello thanks, could you please fix the conflict and merge?

**Pull request merged**
